### PR TITLE
fix(otel): support tag flattening [AIT-9352]

### PR DESCRIPTION
**Pull request by mabdinur**
Ensures otel tags/attribute values with the type List, Tuple and/or set are flattened. This operation is already done by the Datadog UI when displaying tags. However we should also do this in the tracer to better align with other languages. 

The flattening logic must follow this spec: 

```
Array values must decay into as many keys as there are entries in the array. The keys must be mapped by concatenating the outer and inner key values together, recursively, separated by a dot. In pseudo-code:
fn addArrayAttribute(key, array):
for (subkey, value) in array:
addScalarOrArrayAttribute(key + "." + subkey, value)
Example: Given attributes {"key": [[1,2], ["3", "4"]]}, we will have {"key.0.0": "1", "key.0.1": 2, "key.1.0": 3, "key.1.1":4}.
```

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by emmettbutler — COMMENTED** [review]
I'll wait to do a proper review until CI is passing. Let me know if you'd like detailed feedback before then.

**Review by emmettbutler — COMMENTED** [review]
The change looks great. What do you think is causing the performance regression?

File: ddtrace/opentelemetry/_span.py [hunk 1]
```diff
@@ -160,7 +161,11 @@ def set_attribute(self, key, value):
             _ddmap(self._ddspan, ddattribute, value)
             return
 
-        self._ddspan.set_tag(key, value)
+        if is_sequence(value):
+            for k, v in flatten_key_value(key, value).items():
+                self._ddspan.set_tag(k, v)
+            return
+        self._ddspan.set_tag(k, v)
```
Comment (mabdinur) [inline ddtrace/opentelemetry/_span.py:168]:
```suggestion
        self._ddspan.set_tag(key, value)
```

**Review by emmettbutler — APPROVED** [review]
Nice! I see from the benchmarking platform (CI pipeline ID 
26358953) that you solved the performance regression.

**Pull request merged**